service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/bean/FluxMessageServiceBean.java

```diff
@@ -334,41 +334,48 @@ private void updateFishingTripStartAndEndDate(Set<FaReportDocumentEntity> faRepo
     private void updateGeometry(FaReportDocumentEntity faReportDocumentEntity) throws ServiceException {
         List<MicroMovement> movements = getInterpolatedGeomForArea(faReportDocumentEntity);
         Set<FishingActivityEntity> fishingActivityEntities = faReportDocumentEntity.getFishingActivities();
-        List<Geometry> multiPointForFaReport = populateGeometriesForFishingActivities(movements, fishingActivityEntities);
+        List<Geometry> multiPointForFaReport = populateGeometriesForFishingActivitiesAndRelatedActivities(movements, fishingActivityEntities);
         faReportDocumentEntity.setGeom(GeometryUtils.createMultipoint(multiPointForFaReport));
     }
 
-    private List<Geometry> populateGeometriesForFishingActivities(List<MicroMovement> movements, Set<FishingActivityEntity> fishingActivityEntities) throws ServiceException {
+    private List<Geometry> populateGeometriesForFishingActivitiesAndRelatedActivities(List<MicroMovement> movements, Set<FishingActivityEntity> fishingActivityEntities) throws ServiceException {
         List<Geometry> multiPointForFaReport = new ArrayList<>();
-        for (FishingActivityEntity fishingActivity : Utils.safeIterable(fishingActivityEntities)) {
-            List<Geometry> multiPointForFa = new ArrayList<>();
-            Instant activityDate = fishingActivity.getOccurence() != null ? fishingActivity.getOccurence() : getFirstDateFromDelimitedPeriods(fishingActivity.getDelimitedPeriods());
-            Geometry interpolatedPoint = interpolatePointFromMovements(movements, activityDate);
-
-            for (FluxLocationEntity fluxLocation : fishingActivity.getFluxLocations()) {
-                Geometry point = null;
-                String fluxLocationStr = fluxLocation.getTypeCode();
-                if (fluxLocationStr.equalsIgnoreCase(FluxLocationEnum.AREA.name())) {
-                    point = interpolatedPoint;
-                    fluxLocation.setGeom(point);
-                } else if (fluxLocationStr.equalsIgnoreCase(FluxLocationEnum.LOCATION.name())) {
-                    point = getGeometryForLocation(fluxLocation);
-                    log.debug("Geometry calculated for location is: {}", point);
-                    fluxLocation.setGeom(point);
-                } else if (fluxLocationStr.equalsIgnoreCase(FluxLocationEnum.POSITION.name())) {
-                    point = GeometryUtils.createPoint(fluxLocation.getLongitude(), fluxLocation.getLatitude());
-                    fluxLocation.setGeom(point);
-                }
-                if (point != null) {
-                    multiPointForFa.add(point);
-                    multiPointForFaReport.add(point);
-                }
+        for (FishingActivityEntity fishingActivityEntity : Utils.safeIterable(fishingActivityEntities)) {
+            multiPointForFaReport = deriveMultipointsFromActivitiesAndMovements(multiPointForFaReport, fishingActivityEntity, movements);
```

Comment:
Here again I would prefer if we didn't return the multiPointForFaReport list that we send in. Right now it kind of looks like we are "overwriting" it but we are actually not. This method can still end with just "return multiPointForFaReport;".

Reply:
Naaaa, here it is different input and output.

Reply:
You are right, my bad!